src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.ClassificationVerifier.cs

```diff
@@ -0,0 +1,212 @@
+                {
+                    _fontAndColorStorage = _serviceProvider.GetService<SVsFontAndColorStorage, IVsFontAndColorStorage>();
+                    _fontAndColorUtilities = (IVsFontAndColorUtilities)_fontAndColorStorage!;
+                }
+
+                // Make no changes when in high contast mode or in unknown theme.
+                if (SystemParameters.HighContrast ||
+                    !_colorSchemes.TryGetValue(schemeName, out var colorScheme) ||
+                    !colorScheme.ContainsKey(themeId))
+                {
+                    return false;
+                }
+
+                var coreThemeColors = (themeId == KnownColorThemes.Dark)
+                    ? DarkThemeForeground
+                    : BlueLightThemeForeground;
+
+                var colorSchemeTheme = colorScheme[themeId];
+
+                // Open Text Editor category for readonly access and do not load items if they are defaulted.
+                if (_fontAndColorStorage.OpenCategory(TextEditorMEFItemsColorCategory, (uint)__FCSTORAGEFLAGS.FCSF_READONLY) != VSConstants.S_OK)
+                {
+                    // We were unable to access color information.
+                    return false;
+                }
+
+                try
+                {
+                    foreach (var classification in Classifications)
+                    {
+                        var colorItems = new ColorableItemInfo[1];
+
+                        if (_fontAndColorStorage.GetItem(classification, colorItems) != VSConstants.S_OK)
+                        {
+                            // Classifications that are still defaulted will not have entries.
+                            continue;
+                        }
+
+                        var colorItem = colorItems[0];
+
+                        if (IsClassificationCustomized(coreThemeColors, colorSchemeTheme, colorItem, classification))
+                        {
+                            return true;
+                        }
+                    }
+                }
+                finally
+                {
+                    _fontAndColorStorage.CloseCategory();
+                }
+
+                return false;
+            }
+
+            /// <summary>
+            /// Determines if the ColorableItemInfo's Foreground has been customized to a color that doesn't match the
+            /// selected scheme.
+            /// </summary>
+            private bool IsClassificationCustomized(
+                ImmutableDictionary<string, uint> coreThemeColors,
+                ImmutableDictionary<string, uint> schemeThemeColors,
+                ColorableItemInfo colorItem,
+                string classification)
+            {
+                AssertIsForeground();
+
+                var foregroundColorRef = colorItem.crForeground;
+
+                if (_fontAndColorUtilities!.GetColorType(foregroundColorRef, out var foregroundColorType) != VSConstants.S_OK)
```

Comment:
nit: Personal preference is to add `Contract.ThrowIfNull` for assumed non null members at the beginning of a function. It's easier to spot that the assumption is there